client_internals_it_test.go

```diff
@@ -204,6 +205,58 @@ func TestClientInternal_ClusterID(t *testing.T) {
 	assert.Equal(t, types.UUID{}, ci.ClusterID())
 }
 
+func TestClientInternal_OrderedMembers(t *testing.T) {
+	// start a 1 member cluster
+	tc := it.StartNewClusterWithOptions("ci-orderedmembers", 55701, 1)
+	defer tc.Shutdown()
+	ctx := context.Background()
+	client := it.MustClient(hz.StartNewClientWithConfig(ctx, tc.DefaultConfig()))
+	defer client.Shutdown(ctx)
+	ci := hz.NewClientInternal(client)
+	targetUUIDs := tc.MemberUUIDs
+	assert.True(t, sameMembers(targetUUIDs, ci.OrderedMembers()))
+	// start another member
+	mem, err := tc.RC.StartMember(ctx, tc.ClusterID)
+	if err != nil {
+		t.Fatal(err)
+	}
+	targetUUIDs = append(targetUUIDs, mem.UUID)
+	assert.True(t, sameMembers(targetUUIDs, ci.OrderedMembers()))
+	// start another member
+	mem, err = tc.RC.StartMember(ctx, tc.ClusterID)
+	if err != nil {
+		t.Fatal(err)
+	}
+	targetUUIDs = append(targetUUIDs, mem.UUID)
+	assert.True(t, sameMembers(targetUUIDs, ci.OrderedMembers()))
+	// stop a member
+	stopped := ci.OrderedMembers()[0]
+	if _, err := tc.RC.ShutdownMember(ctx, tc.ClusterID, stopped.UUID.String()); err != nil {
+		t.Fatal(err)
+	}
+	targetUUIDs = targetUUIDs[1:]
+	it.Eventually(t, func() bool {
+		return sameMembers(targetUUIDs, ci.OrderedMembers())
+	})
+	types.NewUUID()
+	assert.False(t, ci.ConnectedToMember(stopped.UUID))
+	for _, mem := range ci.OrderedMembers() {
+		assert.True(t, ci.ConnectedToMember(mem.UUID))
+	}
+}
+
+func TestClientInternal_ConnectedToMember(t *testing.T) {
+	tc := it.StartNewClusterWithOptions("ci-orderedmembers", 55701, 2)
```

Comment:
Cluster name is same with the test above, is that intentional? If so can you explain 🙏

Reply:
Is there a reason we test against even number of members?

Reply:
> We can defer shutdown call for the cluster after this in case test exists before the actual Shutdown call. Wdyt?

I think the test cannot exit successfully without calling `Shutdown`. If the test fails, all testing stops, so I think it's OK to leave it as is.